### ISSUE TITLE
Registry credential can not be saved on projects (v2.4.3 )

### ISSUE THREAD

**Issue by mrvarmazyar** [rancher/rancher#27117]
Hey,

I can't save registry credentials in the projects and I face on with this error in inspect element:
```
component.js:79 Uncaught TypeError: Cannot read property 'id' of null
    at n.willSave (component.js:79)
    at n.save (vendor-bb34367e194e1fceca2f1829b04bc1f2.js:20378)
    at e.n._join (vendor-bb34367e194e1fceca2f1829b04bc1f2.js:4658)
    at e.n.join (vendor-bb34367e194e1fceca2f1829b04bc1f2.js:4623)
    at p (vendor-bb34367e194e1fceca2f1829b04bc1f2.js:3473)
    at vendor-bb34367e194e1fceca2f1829b04bc1f2.js:1775
    at s (vendor-bb34367e194e1fceca2f1829b04bc1f2.js:3382)
    at n.save (vendor-bb34367e194e1fceca2f1829b04bc1f2.js:1775)
    at n.save (vendor-bb34367e194e1fceca2f1829b04bc1f2.js:19389)
    at n.send (vendor-bb34367e194e1fceca2f1829b04bc1f2.js:3045)
willSave @ component.js:68
save @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:20378
n._join @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:4658
n.join @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:4623
p @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:3473
(anonymous) @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:1775
s @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:3382
(anonymous) @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:1775
save @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:19389
send @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:3045
n @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:2999
(anonymous) @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:1805
s @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:3382
(anonymous) @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:1805
n._run @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:4659
n._join @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:4658
n.join @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:4623
p @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:3473
t.handler @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:1804
(anonymous) @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:3102
dispatch @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:1014
g.handle @ vendor-bb34367e194e1fceca2f1829b04bc1f2.js:1009
```
Please note that my rancher version is 2.4.3, so tell me know how can I fix it? or if it's a bug on your side, how long does it take to be fixed?

Thank you

**Comment by sangeethah** [comment]
This is a UI issue that prevents registry credentials to be created as project level and will be fixed in the next release - 2.4.4. 

You would be able to create a registry that is scoped to a namespace using UI.

As workaround, you can create a Registry credential at project level using UI API -https://<rancher server>/v3/project/<clusterid>:<projectId>/dockerCredentials
Provide values for the following fields 
```
{
"description": "test1",
"name": "test1",
"registries": {
"<registry>": {
"username": “<username>",
"password": “<password>"
}
```

**Comment by soumyalj** [comment]
Reproduced the issue in v2.4.3
1. Create a registry selecting "Available to all namespaces" and save
UI is stuck in Saving state and does not complete. 
![Screen Shot 2020-05-20 at 11 53 10 AM](https://user-images.githubusercontent.com/18536626/82485578-8379ff00-9a90-11ea-96e2-e725e4a2acf9.png)

Validated with v2.4-head(8e20687a4)
1. Add a registry selecting the scope "Available to all namespaces" .
2. Configure Dockerhub/quay/artifactory/custom registry
3. Save
Registry gets created as expected without any errors. 

Note: Creating registry in a single namespace is also functional.

**Comment by tmpkn** [comment]
> This is a UI issue that prevents registry credentials to be created as project level and will be fixed in the next release - 2.4.4.

I am on 2.4.4 and am still experiencing this error.

**Comment by soumyalj** [comment]
@tmpkn : This fix did not make it to release v2.4.4. It will be available in release v2.4.5.
